### ISSUE TITLE
No support for Imagick formats

### ISSUE THREAD

**Issue by chrismaaz** [ServBay/ServBay#5]
I'm having trouble with Imagick. It doesn't seem to support any file formats.

`print_r( Imagick::queryFormats() );`

Gives an empty array. When running my code to read a PDF file for conversation I get the following error: 

`Warning: Imagick::readImage(): Stream does not support seeking
Uncaught ImagickException: NoDecodeDelegateForThisImageFormat`

Tried both PHP 8.2 and PHP 8.3 and I've made sure the module is loaded. The code was working fine on my previous LAMP installation so I am suspecting that the Imagick implementation of Servbay is pooped or missing some features.

**Comment by naizhao** [comment]
hi, this is a known bug of imagick:https://github.com/Imagick/imagick/issues/695

**Comment by chrismaaz** [comment]
I am running Apple Silicon and as far as I understand the bug only affects Intel machines? Will this be fixed in a upcoming release?

**Comment by naizhao** [comment]
> I am running Apple Silicon and as far as I understand the bug only affects Intel machines? Will this be fixed in a upcoming release?

Just did a quick test on php 8.2, 8.3 and 8.4, seems to be working fine. M2 Chip

![SB-002447](https://github.com/user-attachments/assets/51f6c3a9-0db2-4309-bce5-ca88e5b9d2d5)
![SB-002448](https://github.com/user-attachments/assets/0063cc58-a9b5-48ed-82ff-f48497d0d972)
![SB-002449](https://github.com/user-attachments/assets/8aaac8b1-970b-4ee4-ab72-6852eab5feb2)

**Comment by naizhao** [comment]
```php
<?php
echo phpversion();
print_r( Imagick::queryFormats() );
```

**Comment by naizhao** [comment]
Use phpinfo() to see if this shows 0. If so, that's the bug I mentioned. Until the imagick team releases a fix, recompile the imagick module and that should fix your problem.

![SB-002450](https://github.com/user-attachments/assets/394f455d-f25d-4054-bf76-25aa71670869)

**Comment by chrismaaz** [comment]
<img width="947" alt="Screenshot 2025-01-03 at 18 26 17" src="https://github.com/user-attachments/assets/1b8c1602-88e8-4b04-8805-800b71debf91" />

**Comment by naizhao** [comment]
> <img alt="Screenshot 2025-01-03 at 18 26 17" width="947" src="https://private-user-images.githubusercontent.com/11076809/400029278-1b8c1602-88e8-4b04-8805-800b71debf91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU5MjkwOTMsIm5iZiI6MTczNTkyODc5MywicGF0aCI6Ii8xMTA3NjgwOS80MDAwMjkyNzgtMWI4YzE2MDItODhlOC00YjA0LTg4MDUtODAwYjcxZGViZjkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAzVDE4MjYzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUyZjM5NTgyMGUyOWIzMjBlNGNmZjM1M2QwYTQ5OTU0ODgzYmM1ZWE2OTY2NGQyNTNkMzVlMzc5NWQ1OGJmZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Xf8VMWfb4JaJbjnRZg4bzSyI79HEpASWCtii_iBlsHk">

Yes, it is this bug. The cause of this bug is not clear at present, and it is not 100% reproducible, but it is related to the versions of xcode, clang, and macOS. Upgrading your xcode sdk may solve this problem, or recompiling imagick on your computer may also solve this problem.

**Comment by naizhao** [comment]
@chrismaaz hi, please go to `Services` - `Common Services` then install `ServBay Development Library`, then restart PHP, it should be fixed.

**Comment by chrismaaz** [comment]
Thanks, just out of curiosity, what does the "ServBay Development Library" include? I can't find any documentation what it actually does and what it is needed for?
